pkg/testutil/scenarios.go

```diff
@@ -42,14 +57,23 @@ type PathContentFlag struct {
 	Content func() string
 }
 
+// NewEnvConfig provides the details required to setup a temporary test
```

Comment:
I'm not a fan of the `New` prefix on the struct name as `New` is typically used in go to indicate a constructor function. This struct should just be `EnvConfig` as that's what it represents/contains details of (i.e. an environment config).